content/core-features/authentication.textile

```diff
@@ -373,6 +374,127 @@ The following JavaScript example demonstrates how to issue an "Ably Token":#ably
+
+Token revocation has to be enabled for you by Ably. There is a performance cost to enabling this feature. "Contact Ably":https://ably.com/contact if you'd like to enable this feature for a specific application.
+
+h3(#development-status). Development status
+
+The revocable tokens feature is currently in Beta.
+
+h3. Revocable tokens
+
+Tokens are revocable if they are created from an API key that has a @revocableTokens@ attribute set to @true@. 
+
+Tokens created from a key with the @revocableTokens@ attribute set to @true@ are restricted to a TTL of 1 hour; an attempt to create, or present, a token that is revocable, but has an TTL of longer than 1 hr, is rejected with an error.
+
+A JWT may include the following additional claim:
+
+- @x-ably-revocation-key@ := a string used to identify which token(s) to revoke in the revocation request.
+
+Designating a revocation key for a token, or a group of tokens, enables the revocation process to be used at any level of granularity, depending on the needs of the application.
+
+h3. Token revocation API
+
+Ably provides a token revocation API.
+
+h4(#target-specifiers). Target specifiers
+
+A revocation request must include one or more target specifiers which define the token(s) that are affected by the request. Where multiple target specifiers are included, this is equivalent to making multiple independent revocation requests, each for a single target. The tokens affected by a request are those that match at least one of the target specifiers included in the request.
+
+By default, supported target specifiers are:
+
+- clientId:<clientId> := For a string @clientId@, matches tokens that have a non-empty @clientId@ equal to that given.
+- revocationKey:<revocationKey> := Matches tokens that have a @revocationKey@ claim that matches the given @revocationKey@.
+
+The following target specifier can also be enabled for you by Ably, should your use case require it:
+
+- channel:<resourceName> := For a @resourceName@ that exactly matches one of the "resource names":/core-features/authentication#wildcards present in the token capabilities. Note that this is not the same thing as revoking all tokens that have access to the channel. For example, a token with a capability of @{"foo:*": ["*"]}@ will be revoked by a target of @channel:foo:*@, but a revocation to @channel:*:*@ will have no effect (even though that is a superset of the capabilities of @foo:*@), and nor will @channel:foo:bar@ (even for connections using the token to attach to that particular channel). It must be the exact string used in the token capabilities (which may be inherited from key capabilities).  
+
+h4. API endpoint
+
+The Ably REST API contains the following endpoint for token revocation:
+
+```
+POST /keys/<keyName>/revokeTokens
+```
+
+Where the @keyName@ is @appId.keyId@.
+
+In that request, the request body has the following form:
+
+```[json]
```

Comment:
I'm not sure the comments add much to this code sample given the properties are defined below. I wonder if example values would be better?

Reply:
Removed and expanded properties below where necessary. Replaced by actual values.